src/utils/parser/index.js

```diff
@@ -5,26 +5,29 @@ const { getValue } = require("devtools-config");
 
 let worker;
 
-function restartWorker() {
-  if (worker) {
-    worker.terminate();
-  }
-
+function startWorker() {
   worker = new Worker(`${getValue("baseWorkerURL")}parser-worker.js`);
 }
 
-restartWorker();
-
 function destroyWorker() {
   if (worker) {
     worker.terminate();
     worker = null;
   }
 }
 
-const getSymbols = workerTask(worker, "getSymbols");
-const getVariablesInScope = workerTask(worker, "getVariablesInScope");
-const getExpression = workerTask(worker, "getExpression");
+function getOrCreateWorker() {
+  if (!worker) {
+    startWorker();
+  }
+
+  return Promise.resolve(worker);
+}
+
+const getSymbols = workerTask(getOrCreateWorker, "getSymbols");
+const getVariablesInScope =
+  workerTask(getOrCreateWorker, "getVariablesInScope");
+const getExpression = workerTask(getOrCreateWorker, "getExpression");
 
 module.exports = {
   getSymbols,
```

Comment:
These changes are needed to be able to mock it using jest; the problem is that it was starting the worker unconditionally. Now I made it lazy.

I decided to make `getOrCreateWorker` return a Promise because `workerTask` is already promise-based and we might want to do some async initialization for some workers in the future (like waiting for a "ready" message or something)).